mage_ai/data_integrations/utils/config.py

```diff
@@ -33,6 +38,16 @@ def __get_settings(absolute_file_path, variables: Dict = {}, pipeline: 'Pipeline
             json.dumps(pipeline.data_integration['catalog']),
             variables,
         )
+    elif pipeline and not settings.get('catalog'):
+        catalog_config_path = os.path.join(
+            get_repo_path(),
+            PIPELINES_FOLDER,
+            pipeline.uuid,
+            DATA_INTEGRATION_CATALOG_FILE,
+        )
+        with open(catalog_config_path, 'r') as f:
+            catalog = interpolate_variables(f.read(), variables)
```

Comment:
@wangxiaoyou1993 Though everything seems to run fine when I tested it, this line I was a little unsure if I was calling the  `interpolate_variables` method correctly. Or if `f` versus `f.read()` was the correct first argument to pass.

Reply:
this seems useless. you already read the catalog file and set it in `pipeline.data_integration` when loading the pipeline.